### PR TITLE
fix: add authorization mode id for all traces

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Previously, the authorization model id is not available in the traces of some operations, such as ListObject. This change is to ensure that the traces are available in all operations that requires authorization model.

## References
Close https://github.com/openfga/openfga/issues/499

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -227,7 +227,6 @@ func (s *Server) Check(ctx context.Context, req *openfgapb.CheckRequest) (*openf
 	if err != nil {
 		return nil, err
 	}
-	span.SetAttributes(attribute.KeyValue{Key: "authorization-model-id", Value: attribute.StringValue(modelID)})
```
Comment (jon-whit) [inline pkg/server/server.go:230]: Please make this a `const` at the top of this package.

Comment (adriantam) [reply pkg/server/server.go:230]: Done

**Review by craigpastro — DISMISSED** [review]
LGTM

**Comment by craigpastro** [comment]
Sorry, let me check one thing with trace key names. I remember there are best practices.

File: pkg/server/server.go [hunk 2]
```diff
@@ -25,7 +25,8 @@ import (
 type ExperimentalFeatureFlag string
 
 const (
-	AuthorizationModelIDHeader = "openfga-authorization-model-id"
+	AuthorizationModelIDHeader   = "openfga-authorization-model-id"
+	AuthorizationModelIDTraceTag = "authorization-model-id"
```
Comment (craigpastro) [inline pkg/server/server.go:29]: According to https://opentelemetry.io/docs/reference/specification/common/attribute-naming/, keys should use snake_case. 

> For each multi-word dot-delimited component of the attribute name separate the words by underscores (i.e. use snake_case). For example http.status_code denotes the status code in the http namespace.

Could you update this key, please?

Comment (adriantam) [reply pkg/server/server.go:29]: Done

Comment (craigpastro) [reply pkg/server/server.go:29]: I think you updated the header, and not the trace key, by accident.

Comment (adriantam) [reply pkg/server/server.go]: changed

**Review by craigpastro — APPROVED** [review]
LGTM. Thanks!

**Review by jon-whit — APPROVED** [review]

**Pull request merged**